### ISSUE TITLE
[CI] Failure in RangeQueryBuilderTests.testToQuery 

### ISSUE THREAD

**Issue by cbuescher** [elastic/elasticsearch#28147]
This happend last night on 5.6 branch:

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+5.6+periodic/1253/console

Strangely I cannot reproduce locally yet, although these normally reproduce well. I'll dig into it later.

```
REPRODUCE WITH: ./gradlew :core:test -Dtests.seed=F3EFFE4F7FDF63A4 -Dtests.class=org.elasticsearch.index.query.RangeQueryBuilderTests -Dtests.method="testToQuery" -Dtests.security.manager=true -Dtests.locale=ko-KR -Dtests.timezone=Africa/Lome

22:40:08 ERROR   0.04s J0 | RangeQueryBuilderTests.testToQuery <<< FAILURES!
22:40:08    > Throwable #1: java.lang.IllegalArgumentException: min value (101) is greater than max value (100)
22:40:08    > 	at __randomizedtesting.SeedInfo.seed([F3EFFE4F7FDF63A4:414FC710E5CA64E]:0)
22:40:08    > 	at org.apache.lucene.document.IntRange.verifyAndEncode(IntRange.java:132)
22:40:08    > 	at org.apache.lucene.document.IntRange.encode(IntRange.java:113)
22:40:08    > 	at org.apache.lucene.document.IntRange.newRelationQuery(IntRange.java:235)
22:40:08    > 	at org.apache.lucene.document.IntRange.newIntersectsQuery(IntRange.java:191)
22:40:08    > 	at org.elasticsearch.index.mapper.RangeFieldMapper$RangeType$5.intersectsQuery(RangeFieldMapper.java:677)
22:40:08    > 	at org.elasticsearch.index.mapper.RangeFieldMapper$RangeType.rangeQuery(RangeFieldMapper.java:779)
22:40:08    > 	at org.elasticsearch.index.mapper.RangeFieldMapper$RangeFieldType.rangeQuery(RangeFieldMapper.java:293)
22:40:08    > 	at org.elasticsearch.index.query.RangeQueryBuilder.doToQuery(RangeQueryBuilder.java:509)
22:40:08    > 	at org.elasticsearch.index.query.AbstractQueryBuilder.toQuery(AbstractQueryBuilder.java:96)
22:40:08    > 	at org.elasticsearch.test.AbstractQueryTestCase.testToQuery(AbstractQueryTestCase.java:571)
22:40:08    > 	at java.lang.Thread.run(Thread.java:748)
```

**Comment by cbuescher** [comment]
This is caused when we hit a very rare combination of things in the test: a "from: 100, to: 101" query on an "integer_range" field, with both "includeUpper" and "includeLower" set to false. In this case the range goes from "101 to 100" and the RangeFieldMapper complains. We shouldn't test this combination.
This seems to only happen on 5.6 because on the 6.x and master branch we don't test with "integer_range" anymore. Still need to dig why that happened and if we should do that again.

**Comment by cbuescher** [comment]
> On the 6.x and master branch we don't test with "integer_range" anymore. Still need to dig why that happened and if we should do that again.

Just checked, the reason we don't test this anymore in the core query builder tests is due to moving the RangeFieldMapper and other to the new  `mappers-extra` module in #26549, so we don't need to adapt any tests there.

**Comment by jasontedor** [comment]
> Just checked, the reason we don't test this anymore in the core query builder tests is due to moving the RangeFieldMapper and other to the new  `mappers-extra` module in #26549, so we don't need to adapt any tests there.

The range field mapper was moved back to core in #27854.

**Comment by cbuescher** [comment]
Okay, in that case it might make sense to add the test for the range fields in the range query builder tests. Will do just that.

**Comment by cbuescher** [comment]
The failure on the 5.6 branch is fixed by 771fdc, and #28171 should track re-adding those
tests to master and 6.x, so I'm closing this.
